contrib/libadd_mca_comp_update.py

```diff
@@ -0,0 +1,232 @@
+                print("  ---> Error: Makefile.am does not contain necessary anchor")
+                continue
+
+            #
+            # Sanity Check: This file does not need to be updated.
+            #
+            if True == has_libadd and all(False == v for v in libadd_missing):
+                skipped_files.append("Skip: "+make_filename)
+                print("       Skip: Already updated Makefile.am")
+                continue
+
+            #
+            # Now go though and create a new version of the Makefile.am
+            #
+            r_fd = open(make_filename, 'r')
+            w_fd = open(make_filename+".mod", 'w')
+
+            num_libadds=0
+            for line in r_fd:
+                # LDFLAGS anchor
+                if re.search( r"^\s*"+ldflags_field, line):
+                    w_fd.write(line)
+                    # If there is no LIBADD, then put it after the LDFLAGS
+                    if False == has_libadd:
+                        w_fd.write(libadd_field+" = "+libadd_str+"\n")
+                # Existing LIBADD field to extend
+                elif 0 == num_libadds and re.search( r"^\s*"+libadd_field, line):
+                    parts = line.partition("=")
+                    num_libadds += 1
+
+                    if parts[0][-1] == '+':
+                        w_fd.write(libadd_field+" += ")
+                    else:
+                        w_fd.write(libadd_field+" = ")
+
+                    # If all libs are missing, then add the full string
+                    # Otherwise only add the missing items
+                    if all(True == v for v in libadd_missing):
+                        w_fd.write(libadd_str)
+                        # Only add a continuation if there is something to continue
+                        if 0 != len(parts[2].strip()):
+                            w_fd.write(" \\")
+                        w_fd.write("\n")
+                    else:
+                        partly_files.append("Partly updated: "+make_filename)
+                        for idx, lib in enumerate(libadd_list):
+                            if True == libadd_missing[idx]:
+                                w_fd.write(lib+" \\\n")
+
+                    # Original content (unless it's just a line continuation)
+                    if 0 != len(parts[2].strip()) and parts[2].strip() != "\\":
+                        w_fd.write("\t"+parts[2].lstrip())
+
+                # Non matching line, just echo
+                else:
+                    w_fd.write(line)
+
+            r_fd.close()
+            w_fd.close()
+
+            #
+            # Replace the original with the updated version
+            #
+            shutil.move(make_filename+".mod", make_filename)
+            updated_files.append(make_filename)
+
+        
+if __name__ == "__main__":
+
+    update_makefile_ams()
+
+    print("")
+
+    print("="*40);
+    print("{:>3} : Files skipped".format(len(skipped_files)))
```

Comment:
Probably just need the `len(` version here since I'm using the new `.format` mode that doesn't need it to be converted to a string first. I'll update the others.